### PR TITLE
More specific error code in set response

### PR DESCRIPTION
**Pull request by KarloX2**
In case an ArgumentException is caught while applying the data of a set request, try parse the exception's message and derive the ErrorCode for the set response from it, if possible (if not possible fallback to ErrorCode.WrongType like it was before this change).

### CODE REVIEW

**Comment by lextm** [comment]
Can you add a test case similar to the TimeOut one, https://github.com/lextm/sharpsnmplib/blob/master/Tests/CSharpCore/Integration/DaemonTestFixture.cs#L693 ?

That can illustrate why this change is needed.

**Comment by KarloX2** [comment]
Ok, I added a test case. The intention of this change is to have a way to reply with the appropriate error code to the sender of a set response, instead of just always sending ErrorCode.WrongType in case the variable's setter throws an ArgumentException.

File: SharpSnmpLib/Objects/SysContact.cs [hunk 1]
```diff
@@ -62,6 +62,10 @@ public override ISnmpData Data
                 {
                     throw new ArgumentException("Invalid data type.", nameof(value));
                 }
+                if (((OctetString)value).ToString().Length > 255) //respect DisplayString syntax length limitation
```
Comment (lextm) [inline SharpSnmpLib/Objects/SysContact.cs]: Instead of modifying this class, you should follow the TimeOut test case manner to create a dedicate class for testing in the unit testing project.

File: SharpSnmpLib/Pipeline/SetMessageHandler.cs [hunk 2]
```diff
@@ -1,114 +1,115 @@
...
-                }
-
-                result.Add(v);
-            }
-
-            context.GenerateResponse(result);
-        }
-    }
-}
+﻿// SET message handler class.
```
Comment (lextm) [inline SharpSnmpLib/Pipeline/SetMessageHandler.cs]: Did you change line endings of this file?

File: Tests/CSharpCore/Integration/DaemonTestFixture.cs [hunk 3]
```diff
@@ -616,7 +616,7 @@ public void TestResponsesFromSingleSourceWithMultipleThreads(int count)
                             {
                                 GetRequestMessage message = new GetRequestMessage(index, VersionCode.V2,
                                     new OctetString("public"),
-                                    new List<Variable> {new Variable(new ObjectIdentifier("1.3.6.1.2.1.1.1.0"))});
+                                    new List<Variable> { new Variable( new ObjectIdentifier("1.3.6.1.2.1.1.1.0")) });
```
Comment (lextm) [inline Tests/CSharpCore/Integration/DaemonTestFixture.cs]: Can you revert the changes to this line?

**Comment by lextm** [comment]
Looks good. But I don't want to add a length limit to the actual class right now. Let's create a test class instead.

File: SharpSnmpLib/Pipeline/SetMessageHandler.cs (hunk 2 above)
Comment (KarloX2) [reply SharpSnmpLib/Pipeline/SetMessageHandler.cs]: Maybe inadvertently. I'll check.

File: Tests/CSharpCore/Integration/DaemonTestFixture.cs (hunk 3 above)
Comment (KarloX2) [reply Tests/CSharpCore/Integration/DaemonTestFixture.cs]: Ok

**Comment by KarloX2** [comment]
Hi lextm! I modified the test case in DaemonTestFixture.cs like you suggested (I think). I tried a lot to revert the changes made to SysContact.cs and to fix the line ending problem, but my knowledge about GitHub isn't good enough. I don't understand all those reset/rebase/checkout/sync/blame/diff/branch etc. functions well enough. I'll certainly have to invest some time to learn all this, but currently I can't, sorry.

Therefore, are you able to pick out only the changes needed in DaemonTestFixture.cs and SetMessageHandler.cs? Thanks!

**Comment by lextm** [comment]
It is OK. I will accept the PR and perform some cleanup afterwards.
